examples/game/breakout.rs

```diff
@@ -1,16 +1,54 @@
+//! A simplified implementation of the classic game "Breakout"
+
 use bevy::{
     core::FixedTimestep,
+    math::{const_vec2, const_vec3},
     prelude::*,
     sprite::collide_aabb::{collide, Collision},
 };
 
-/// An implementation of the classic game "Breakout"
+// Sets the amount of time that should elapse between each physics step
 const TIME_STEP: f32 = 1.0 / 60.0;
+
+// These constants are in `Transform` units:
+// with the default 2D camera they correspond 1:1 with screen pixels
+// These macros are needed as functions that operate on floats cannot be constant in Rust :(
+const PADDLE_HEIGHT: f32 = -215.0;
+const PADDLE_SIZE: Vec3 = const_vec3!([120.0, 30.0, 0.0]);
+const PADDLE_SPEED: f32 = 500.0;
+const PADDLE_BOUNDS: f32 = 380.0;
+
+// We set the z-value of the ball to 1 so it renders on top in the case of overlapping sprites
+const BALL_STARTING_POSITION: Vec3 = const_vec3!([0.0, -50.0, 1.0]);
+const BALL_SIZE: Vec3 = const_vec3!([30.0, 30.0, 0.0]);
+const BALL_SPEED: f32 = 400.0;
+// This vector is later extended into three dimensions and then normalized
+const INITIAL_BALL_DIRECTION: Vec2 = const_vec2!([0.5, -0.5]);
+
+const PLAY_AREA_BOUNDS: Vec2 = const_vec2!([900.0, 600.0]);
+const WALL_THICKNESS: f32 = 10.0;
+
+const BRICK_ROWS: isize = 4;
+const BRICK_COLUMNS: isize = 5;
+const BRICK_SPACING: f32 = 20.0;
+const BRICK_SIZE: Vec3 = const_vec3!([150.0, 30.0, 1.0]);
+
+const SCOREBOARD_FONT_SIZE: f32 = 40.0;
+const SCOREBOARD_TEXT_PADDING: Val = Val::Px(5.0);
+
+const BACKGROUND_COLOR: Color = Color::rgb(0.95, 0.95, 0.95);
```

Comment:
These colors choices feel worse to me than the original. I don't think we can use the built in colors here as they tend to be too saturated. Maybe try taking a palete from something like https://www.colourlovers.com/palettes or use a color picker like https://color.adobe.com/create/color-wheel

Reply:
My preference was for "ease of reading the code" over "pretty". IMO it's important to use built-in constants here, because the name lets readers connect what they see to the code more easily.